src/main/scala/org/phenoscape/kb/Graph.scala

```diff
@@ -35,14 +36,40 @@ object Graph {
     )
   }
 
-  private def buildPropertyNeighborsQuerySubject(focalTerm: IRI, property: IRI): Query = {
-    val classRelation = NamedRestrictionGenerator.getClassRelationIRI(property)
-    select_distinct('term, 'term_label) from "http://kb.phenoscape.org/" where bgp(
-      t('existential_node, classRelation, focalTerm),
-      t('existential_node, rdfsSubClassOf, 'existential_superclass),
-      t('existential_superclass, classRelation, 'term),
-      t('term, rdfsLabel, 'term_label)
-    )
+  private def buildPropertyNeighborsQuerySubject(focalTerm: IRI, property: IRI, direct: Boolean): Query = {
+    val allNeighbors =
+      sparql"""
+              SELECT DISTINCT ?term ?term_label
+              FROM $KBMainGraph
+              FROM $KBClosureGraph
+              FROM $KBRedundantRelationGraph
+              WHERE {
+                              
+              $focalTerm $property ?term .
+              ?term $rdfsLabel ?term_label .
+              """
+
+    val filterIndirectNeighbors =
+      sparql"""
+              FILTER NOT EXISTS {
+                  $focalTerm $property ?other_term .
+                  ?other_term $rdfsSubClassOf ?term .
+                  FILTER(?other_term != ?term)
+                }
+              
+              FILTER NOT EXISTS {
+                  $property $rdfType $transitiveProperty .
+                  $focalTerm $property ?other_term .
+                  ?other_term $property ?term .
+                  FILTER(?other_term != ?term)
+                }
+              }
+            """
+
+    val filters = if (direct) sparql"$filterIndirectNeighbors" else sparql""
+    val query = sparql"$allNeighbors" + sparql"$filters"
```

Comment:
I don't think this will produce a valid query if `direct` is `false`. There will be a missing bracket at the end. I think you should instead first create `filters` (including either the filters or empty query text) and then embed it in place in the main query.

Reply:
Sorry, I had fixed this but I had forgotten to push my changes.